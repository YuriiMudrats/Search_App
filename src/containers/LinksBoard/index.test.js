import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";

import { LinksBoard } from './index.jsx';
import lists from '../../../test/mockList';

describe('<LinksBoard />', () => {
  it('should return on pending <div>Loading...</div>', () => {
    const renderedComponent = shallow(<LinksBoard pending={true} notFound={false} lists={[]}/>);
    expect(renderedComponent.html()).toEqual('<div>Loading...</div>')
  });
  it('should return on not found <div>Not found</div>', () => {
    const renderedComponent = shallow(<LinksBoard pending={false} notFound={true} lists={[]}/>);
    expect(renderedComponent.html()).toEqual('<div>Not found</div>')
  });
  it('should render an <div> tag', () => {
    const renderedComponent = shallow(<LinksBoard pending={false} notFound={false} lists={lists}/>);
    expect(renderedComponent.type()).toEqual('div');
  });
  it('<div> should have className attribute', () => {
    const renderedComponent = shallow(<LinksBoard pending={false} notFound={false} lists={lists}/>);
    expect(renderedComponent.prop('className')).toBeDefined();
  });
  it('render component should have className attribute', () => {
    const renderedComponent = shallow(<LinksBoard pending={false} notFound={false} lists={lists}/>);
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
