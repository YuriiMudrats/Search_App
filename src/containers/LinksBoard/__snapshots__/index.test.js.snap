// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinksBoard /> render component should have className attribute 1`] = `
<div
  className="refsDiv"
>
  <LinksList
    id={1695}
    key="1695"
    link="Amazons"
  />
  <LinksList
    id={1701}
    key="1701"
    link="Amazon River"
  />
  <LinksList
    id={48139}
    key="48139"
    link="Amazon rainforest"
  />
  <LinksList
    id={90451}
    key="90451"
    link="Amazon (company)"
  />
  <LinksList
    id={6891537}
    key="6891537"
    link="Amazon Video"
  />
  <LinksList
    id={14312829}
    key="14312829"
    link="Amazon Kindle"
  />
  <LinksList
    id={29621629}
    key="29621629"
    link="Amazon"
  />
  <LinksList
    id={41933064}
    key="41933064"
    link="List of original programs distributed by Amazon"
  />
  <LinksList
    id={51442811}
    key="51442811"
    link="Amazon Obhijaan"
  />
  <LinksList
    id={52791355}
    key="52791355"
    link="Amazon Prime"
  />
</div>
`;
